### ISSUE TITLE
Slash key mismatch on Mac client

### ISSUE THREAD

**Issue by meastman** [symless/synergy#5357]
When using client version 1.7.6, everything works fine. When using client version 1.8.2, slash (`/`) does not work correctly. In some programs, it does nothing; in others it becomes other characters, like `^^` or `<RS>`.

Server
OS: Mac OS X 10.11
Version: 1.7.6 (also tried 1.8.2-beta-8da57cd)

Client
OS: Mac OS X 10.11
Version: 1.8.2-beta-8da57cd
Build Date : Mon May 30 2016

Client logs:
```
[2016-05-30T10:56:58] DEBUG1: recv key down id=0x0000002f, mask=0x0000, button=0x002d
[2016-05-30T10:56:58] DEBUG1: mapKey 002f (47) with mask 0000, start state: 0000
[2016-05-30T10:56:58] DEBUG1: find best:  0000 0000
[2016-05-30T10:56:58] DEBUG1: best key index 3 of 3 (exact)
[2016-05-30T10:56:58] DEBUG1: found key in group 0
[2016-05-30T10:56:58] DEBUG1: state: 0000,0000,0000
[2016-05-30T10:56:58] DEBUG1: flip: 0000 (0000 vs 0000 in 0000 - 0000)
[2016-05-30T10:56:58] DEBUG1: desired state: 0000 0000,0000,0000
[2016-05-30T10:56:58] DEBUG1: flip: 0000 (0000 vs 0000 in ffff - 6020)
[2016-05-30T10:56:58] DEBUG1: mapped to 04e, new state 0000
[2016-05-30T10:56:58] DEBUG1: keystrokes:
[2016-05-30T10:56:58] DEBUG1:   button=0x004e virtualKey=0x004d keyDown=down client=0x0000
[2016-05-30T10:56:58] DEBUG1: recv key up id=0x00000000, mask=0x0000, button=0x002d
[2016-05-30T10:56:58] DEBUG1: keystrokes:
[2016-05-30T10:56:58] DEBUG1:   button=0x004e virtualKey=0x004d keyDown=up client=0x0000
```

**Comment by XinyuHou** [comment]
@meastman 
What keyboard layout and language input do you use?

**Comment by meastman** [comment]
> What keyboard layout and language input do you use?

@XinyuHou On both client and server, I use ANSI (QWERTY) and US English. The only keyboard-related thing that I've customized is I've swapped two of the modifier keys (alt/option and win/cmd) for the USB keyboard attached to the server.

**Comment by nlyan** [comment]
@meastman: Can you clarify whether this issue only occurs when the modifier keys are swapped, or regardless? Thanks

**Comment by dpron** [comment]
I see the same thing - with 1.8.2-beta client (on OS X 10.11), server 1.8.2-beta on win 10x64.  When I switch back to an earlier client version (1.8.0-beta), the slash works.  I have no modifier keys configured in Synergy.

**Comment by meastman** [comment]
> Can you clarify whether this issue only occurs when the modifier keys are swapped, or regardless? Thanks

I see it regardless.

**Comment by XinyuHou** [comment]
@meastman 
What language inputs do you have on your system?

**Comment by XinyuHou** [comment]
@dpron 
What keyboard layout and language inputs do you use?

**Comment by meastman** [comment]
> What language inputs do you have on your system?

Just U.S.

<img width="256" alt="screen shot 2016-06-10 at 23 21 43" src="https://cloud.githubusercontent.com/assets/2059468/15983098/2114860c-2f62-11e6-8168-e7e2b097e42a.png">

**Comment by dpron** [comment]
Same here, standard US language.

**Comment by athairus** [comment]
Same issue here. My layout is the US keyboard, nothing special (besides a few mouse enhancing mods?), fresh install of 10.11. Here's the result of a manual bisect:

8da57cdb7f019fb3c03047c3b14c13fda160d157 BAD
dbd08cae9ed11431f9610b84b29c793708245054 BAD
4208e89eaec211d332af9ac7b977107d6712e8dd BAD
177c2764259c097d0f0f0fa6e081c42bcef91ce5 ?
3de98a2f44a20449468496d47f5642b534a80cf6 ?
324e6803ad1ead7eaf36b737fa91d1dbef9d8296 ?
51a18dc9a29c186f2adc32218fb8670f53106b52 ?
239aaf40e6b1b0e41eb6eed0601ea9a1bfcf8557 ?
b5e212757635276f4805eea85decb8c73d9786fc GOOD
b25ed991d9018ac5b86fe307c89e7cfc700f2dfa GOOD

Couldn't find builds for OS X 10.11 for those marked with a "?". I've narrowed it down to 5. Should be an easy bisect from there. Good luck!

**Comment by millerlarson** [comment]
I began noticing this as well after upgrading all my systems to **synergy-v1.8.3-beta-723a8a9** yesterday.

Server: Linux
Clients: Windows 7, OSX 10.11.3

Nothing special about keyboard/lang config.  QWERTY/US all around.  On OSX, "/" does not work, however "?" works just fine.

**Comment by XinyuHou** [comment]
This is probably caused by the fixed in #2765

**Comment by MisterDenis** [comment]
I also started having key mismatches after updating to the nightly in #2765 but I can confirm that it worked flawlessly until a few minutes ago, which is very odd because I made no changes to Synergy on either sides, nor did I do anything aside from restarting my client.

Right now, the "x" maps as a "," (comma), the "/" returns no key input and (weirdly) the spacebar returns a "'" (apostrophe) on the first key press, then a normal space on subsequent... very strange.

**Comment by sim590** [comment]
Client: Mac OSX 10.11
Server: GNU/Linux

I seem to experience a similar issue here, but with canadian multilangual keyboard layout on both client and server. Also, slash **does work** and **space pressed on the server is replaced by a slash on the client**.

**Comment by XinyuHou** [comment]
Merged into #2765

**Comment by alquimista13** [comment]
THANKS GUYS!!!!! versión issue2765-spacebar-issue-beta-b57e8ad works perfect for me, all issue relate to latín MS keyboard was fide with this version.

I can use the full aplicación now. THANKS
